### PR TITLE
Content Library: VM Template support

### PR DESCRIPTION
**Pull request by dougm**
- Add vapi and vcsim support for VM Template (aka vmtx) related APIs

- Add govc library.clone command

- Add vmtx support to govc library.deploy command

### CODE REVIEW

File: govc/cli/register.go [hunk 1]
```diff
@@ -16,11 +16,18 @@ limitations under the License.
...
+import "os"
+
 var commands = map[string]Command{}
 
 var aliases = map[string]string{}
 
-func Register(name string, c Command) {
+var hideUnreleased = os.Getenv("GOVC_SHOW_UNRELEASED") != "true"
+
+func Register(name string, c Command, unreleased ...bool) {
```
Comment (corrieb) [inline govc/cli/register.go]: Would be good to have a comment here explaining the purpose of unreleased. It's not at all obvious

File: govc/library/checkout.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+func (cmd *checkout) Usage() string {
+	return "PATH NAME"
+}
+
+func (cmd *checkout) Description() string {
+	return `Check out Content Library item PATH to vm NAME.
+
+Examples:
+  govc library.checkout -cluster my-cluster my-content/template-vm-item my-vm`
```
Comment (corrieb) [inline govc/library/checkout.go]: Feels to me like the template parameter should maybe have a flag - multiple unflagged trailing parameters feel somewhat hard to reason about. But I'd prefer consistency with other options. Up to you.

File: govc/library/checkout.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+				Folder: folder.Reference().Value,
+			},
+		}
+		if pool != nil {
+			spec.Placement.ResourcePool = pool.Reference().Value
+		}
+		if host != nil {
+			spec.Placement.Host = host.Reference().Value
+		}
+		if cluster != nil {
```
Comment (corrieb) [inline govc/library/checkout.go]: Don't you require a cluster? Where would it go if Cluster isn't specified?

File: govc/library/clone.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+func (cmd *clone) Usage() string {
+	return "PATH NAME"
+}
+
+func (cmd *clone) Description() string {
+	return `Clone VM to Content Library PATH.
+
+Examples:
+  govc library.clone -vm template-vm my-content template-vm-item`
```
Comment (corrieb) [inline govc/library/clone.go]: Same comment as above about multiple freeform parameters that depend on ordering.

File: govc/library/deploy.go [hunk 5]
```diff
@@ -168,35 +169,68 @@ func (cmd *deploy) Run(ctx context.Context, f *flag.FlagSet) error {
...
+			}
 
-		ref, err := m.DeployLibraryItem(ctx, item.ID, deploy)
-		if err != nil {
-			return err
+			deploy := vcenter.DeployTemplate{
+				Name:          name,
+				Description:   cmd.Options.Annotation,
+				DiskStorage:   storage,
+				VMHomeStorage: storage,
```
Comment (corrieb) [inline govc/library/deploy.go]: Would it ever make sense for the DiskStorage and VMHomeStorage to be in separate locations?

File: vapi/simulator/simulator.go [hunk 6]
```diff
@@ -1490,3 +1498,219 @@ func (s *handler) libraryItemDeployID(w http.ResponseWriter, r *http.Request) {
...
+
+	var spec struct {
+		vcenter.DeployTemplate `json:"spec"`
+	}
+	if !s.decode(r, w, &spec) {
+		return
+	}
+
+	if r.Method == http.MethodGet {
+		t := &vcenter.TemplateInfo{ /*TODO*/ }
```
Comment (corrieb) [inline vapi/simulator/simulator.go]: Could this TODO be more descriptive?

File: govc/library/checkout.go (hunk 2 above)
Comment (dougm) [reply govc/library/checkout.go]: There are a few library commands, and a few others in govc that take 2 args when they are both required.  This one and the others are similar to the `cp` command: `cp SOURCE DEST`

File: govc/library/checkout.go (hunk 3 above)
Comment (dougm) [reply govc/library/checkout.go]: One of `pool`, `host` or `cluster` is required for placement.  The error message from vCenter makes that clear if none are set.

File: govc/library/deploy.go (hunk 5 above)
Comment (dougm) [reply govc/library/deploy.go]: Maybe, but we don't support splitting them out in the soap based commands such as `vm.create` and `vm.clone`.  Never had a request to support that, so kept em the same here too.

File: vapi/simulator/simulator.go (hunk 6 above)
Comment (dougm) [reply vapi/simulator/simulator.go]: Done

File: govc/cli/register.go (hunk 1 above)
Comment (dougm) [reply govc/cli/register.go]: Done

**Pull request merged**